addons/promise/fnc_init.sqf

```diff
@@ -0,0 +1,28 @@
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_promise_init
+
+Description:
+    Initializes the required variabels for the promise system.
+
+Parameters:
+    Nothing
+
+Returns:
+    Nothing
+
+Example:
+    (begin example)
+        [] call CBA_fnc_promise_init;
+    (end)
+
+Author:
+    X39
+---------------------------------------------------------------------------- */
+#include "script_component.hpp"
+isNil {
+    if isNil QGVAR(requests) then {
+        GVAR(requests) = [];
+    };
+};
+
+nil
```

Comment:
This function seems like bloat to me. Just run this in preInit and tell people to add requiredAddons if they really need this in preInit as well.

Reply:
I am not aware of how to realize that with the macros in CfgFunctions.hpp `PATHTO_FNC(promise_init);`

Reply:
Just put it in XEH preInit.